numba/cuda/decorators.py

```diff
@@ -129,4 +129,8 @@ def autojitwrapper(func):
 
 def declare_device(name, sig):
     argtypes, restype = sigutils.normalize_signature(sig)
+    if restype is None:
+        msg = 'Return type must be provided for device declarations'
+        raise TypeError(msg)
```

Comment:
I think a normal type error is appropriate here because at this point the error is related to usage of the API with an incorrect type, and not related to compilation.

Reply:
Agree, thanks.